tests/test_funcs.py

```diff
@@ -155,6 +157,100 @@ def test_asdict_preserve_order(self, cls):
         assert [a.name for a in fields(cls)] == list(dict_instance.keys())
 
 
+class TestAsTuple(object):
+    """
+    Tests for `asdict`.
+    """
+    @given(st.sampled_from(IMMUTABLE_SEQ_TYPES))
```

Comment:
You can use SEQUENCE_TYPES here, a list is a valid tuple_factory and something that might be commonly used.